### ISSUE TITLE
Can't copy inside Ansible

### ISSUE THREAD

**Issue by renankaic** [ansible/ansible#59682]
##### SUMMARY

I'm trying to copy the file "/vagrant/configs/mysqld.cnf" to "/etc/mysql/mysql.conf.d/mysqld.cnf" but it's giving me this error:

"An exception occurred during task execution. To see the full traceback, use -vvv. The error was: If you are using a module and expect the file to exist on the remote, see the remote_src option
fatal: [192.168.15.20]: FAILED! => {"changed": false, "msg": "Could not find or access '/vagrant/configs/mysqld.conf' on the Ansible Controller.\nIf you are using a module and expect the file to exist on the remote, see the remote_src option"}"

I have this code in my "playbook.yml" file:

```
- name: "Copy mysqld.conf file"
      copy:       
       src: /vagrant/configs/mysqld.cnf
       dest: /etc/mysql/mysql.conf.d/mysqld.cnf
       owner: root
       group: root
       mode: 0644
      become: yes

```
When I use "vagrant ssh" to access the machine that I'm trying to use Ansible and use the "ls -l /vagrant/configs/" command, I get this result:

vagrant@ubuntu-bionic:~$ ls -l /vagrant/configs
total 16
drwxrwxrwx 1 vagrant vagrant 4096 Jul 27 03:02 ansible
-rwxrwxrwx 1 vagrant vagrant  396 Jul 27 03:00 id_bionic.pub
drwxrwxrwx 1 vagrant vagrant 4096 Feb 22 19:34 manifests
-rwxrwxrwx 1 vagrant vagrant 3049 Feb 22 17:55 mysqld.cnf

In my hosts file I have this code:

```
[mysqlserver]
192.168.15.20

[mysqlserver:vars]
ansible_user=vagrant
ansible_ssh_private_key_file=/home/renan/ambiente_dev/bionic/id_bionic
ansible_python_interpreter=/usr/bin/python3
ansible_ssh_common_args='-o StrictHostKeyChecking=no'
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

Ansible playbook with Vagrant

##### ANSIBLE VERSION

```
ansible 2.8.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/renan/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.16 (default, Apr  6 2019, 01:42:57) [GCC 8.3.0]
```

##### CONFIGURATION

Shows nothing

##### OS / ENVIRONMENT

Ubuntu 19.04 Disco Dingo - Wireless Network

##### STEPS TO REPRODUCE

1 - Install vagrant, virtualbox 6 and ansible in your Ubuntu machine.

2 - Create this Vagrantfile inside a folder of your preference:
```
Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu/bionic64"

  config.vm.define "mysqlserver" do |mysqlserver|
    mysqlserver.vm.network "public_network", ip: "192.168.15.20"    
    mysqlserver.vm.provision "shell", inline: "cat /vagrant/configs/id_bionic.pub >> .ssh/authorized_keys" 
  end
end
```

3 - Use "`ssh-keygen -t rsa -f id_bionic`" command to create keys files without password.

4 - Use "m`kdir configs && mv id_bionic.pub ./configs/`" command

5 - Use "`touch ./configs/mysqld.cnf`" to create an empty MySQL cnf file. Or use your own mysqld.cnf file.

6 - Use "vagrant up" command

7 - Use "`vagrant ssh`" command and when inside the VM use "`ls -l /vagrant/configs/`" command to see the mysqld.cnf file inside the directory

8 - Exit from the VM and use the command create a hosts file inside a ./configs/ansible folder with this content:
```
[mysqlserver]
192.168.15.20

[mysqlserver:vars]
ansible_user=vagrant
ansible_ssh_private_key_file=/[your_Vagrantfile_path]/id_bionic
ansible_python_interpreter=/usr/bin/python3
ansible_ssh_common_args='-o StrictHostKeyChecking=no'
```

9 - Create an playbook.yml file with the following content:
 ```
 - hosts: all
    tasks:
    - name: "Install a MySQL Server"
      apt:
        update_cache: yes
        cache_valid_time: 3600
        name: ["mysql-server-5.7", "python3-mysqldb"]
        state: latest
      become: yes
   - name: "Copy mysqld.conf file"
      copy:       
       src: /vagrant/configs/mysqld.conf
       dest: /etc/mysql/mysql.conf.d/mysqld.cnf
       owner: root
       group: root
       mode: 0644
      become: yes
```

10 - Run `ansible-playbook -i ./configs/ansible/hosts ./configs/ansible/playbook.yml` command and the exception that I mentioned will throw.

 ```
 - hosts: all
    tasks:
    - name: "Install a MySQL Server"
      apt:
        update_cache: yes
        cache_valid_time: 3600
        name: ["mysql-server-5.7", "python3-mysqldb"]
        state: latest
      become: yes
   - name: "Copy mysqld.conf file"
      copy:       
       src: /vagrant/configs/mysqld.conf
       dest: /etc/mysql/mysql.conf.d/mysqld.cnf
       owner: root
       group: root
       mode: 0644
      become: yes
```

##### EXPECTED RESULTS

I expect that the copy command works.

##### ACTUAL RESULTS

It's simple gives an error

```
FAILED! => {"changed": false, "msg": "Could not find or access '/vagrant/configs/mysqld.conf' on the Ansible Controller.\nIf you are using a module and expect the file to exist on the remote, see the remote_src option"}
```

**Comment by ansibot** [comment]
Files identified in the description:
None

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by cpmills1975** [comment]
Correct me if I'm wrong, but it looks like you are trying to run this on the machine running vagrant and not on the vm? The copy module expects to copy a file FROM the ansible controller (the machine you are running vagrant on) TO the destination machine (in your case the VM).  If you already have a config file ON the VM and you simply want to copy it elsewhere ON the VM, see the "remote_src" option as suggested by the error message.

**Comment by mkrizek** [comment]
needs_info

**Comment by renankaic** [comment]
> Correct me if I'm wrong, but it looks like you are trying to run this on the machine running vagrant and not on the vm? The copy module expects to copy a file FROM the ansible controller (the machine you are running vagrant on) TO the destination machine (in your case the VM). If you already have a config file ON the VM and you simply want to copy it elsewhere ON the VM, see the "remote_src" option as suggested by the error message.

**You were right**. I was pointing to the VM source instead of my own linux path.

I changed the "src: /vagrant/configs/mysqld.cnf" to "src: /home/renan/dev_environment/bionic64/configs/mysqld.cnf" and the copy process worked.

About the "remote_src" option: I tried to enable it but it didn't work too. It gaves me the same message. 

But, it's solved. 

I did this mistake because I was following the path that was mentioned in a Ansible course that I'm doing. They are teaching Vagrant + Ansible inside an Windows environment and they had to create a Vagrant VM with Ansible installed on it. But I didn't need to do that because I'm on Linux due to an issue that didn't allow me to finish it using Windows.

Thank you.
